### PR TITLE
Rabbit IFF

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

Laser beams used by rabbits now have IFF built in.

## Why It's Good For The Game

No more friendly fire.

https://user-images.githubusercontent.com/53223414/201524769-00a3c4cb-6021-4844-b150-a61d07e864d9.mp4

### CODE REVIEW

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -28,6 +28,13 @@
 	wound_bonus = -100
 	bare_wound_bonus = -100
 
+/obj/projectile/beam/laser/prehit_pierce(atom/A)
```
Comment (Kitsunemitsu) [inline code/modules/projectiles/projectile/beams.dm:31]: Isn't it more processor efficient to do it the way Loyalty does?

Comment (EgorDinamit) [reply code/modules/projectiles/projectile/beams.dm:31]: The way loyalty does it is questionable.

Comment (Kitsunemitsu) [reply code/modules/projectiles/projectile/beams.dm:31]: Not really, this always checks 1 variable, sets one variable, then if it hits something it checks one flag, and if it pierces it checks another flag and returns Loyalty just checks one flag whether it hits or misses, then sets one flag if it hits. It's less lines of code and less processor intensive.

Comment (EgorDinamit) [reply code/modules/projectiles/projectile/beams.dm:31]: You'd like to think so, but I really don't see anything good about the way you did it.

**Review by dsmith328 — APPROVED** [review]
Looks real good, no more piles of 5 rabbits

**Pull request merged**